### PR TITLE
Easier access to JavaBean style object properties

### PR DESCRIPTION
**Pull request by coconutpalm**
Also 
- fixed bug and added unit test for trace macro
- added io/with-out-string; needed for above tests

### CODE REVIEW

File: src/clj_foundation/io.clj [hunk 1]
```diff
@@ -6,21 +6,31 @@
...
-  (:import [java.io InputStream File ByteArrayInputStream]
+  (:import [java.io InputStream File ByteArrayInputStream ByteArrayOutputStream]
            [java.net URI URL Socket]))
 
 
 ;; From the deprecated clojure.contrib.io library
 ;; https://github.com/clojure/clojure-contrib/blob/b6c6f0fa165975c416c7d135d1a844353527493b/src/main/clojure/clojure/contrib/io.clj#L352
 (defmacro with-out-writer
-  "Opens a writer on f, binds it to *out*, and evalutes body.
+  "Opens a writer on f, binds it to *out*, and evaluates body.
```
Comment (coconutpalm) [inline src/clj_foundation/io.clj:16]: Just a typo

**Comment by gsavovski** [comment]
👍

File: src/clj_foundation/oo.clj [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   :address.state       \"AZ\"
+   :address.postal-code \"99999\"}"
+  [object & props]
+  (->> (if (seq props) props [])
+       (map (fn [prop] [(keywordize (prop-name prop)) `(.. ~object ~@(getter-seq prop) )]))
+       (into {})))
+
+
+#_(defmacro object
+  "Create the Java object specified by clazz using the specified (optional) ctor-args vector as
```
Comment (coconutpalm) [inline src/clj_foundation/oo.clj]: The following two functions are spec'd but not yet implemented.

File: src/clj_foundation/oo.clj [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+   :address.city        \"Acme\"
+   :address.state       \"AZ\"
+   :address.postal-code \"99999\"}"
+  [object & props]
+  (->> (if (seq props) props [])
+       (map (fn [prop] [(keywordize (prop-name prop)) `(.. ~object ~@(getter-seq prop) )]))
+       (into {})))
+
+
+#_(defmacro object
```
Comment (coconutpalm) [inline src/clj_foundation/oo.clj:47]: The following two are spec'd but not yet implemented.

**Comment by greglakomy** [comment]
👍

**Pull request merged**